frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/utils.ts

```diff
@@ -43,6 +43,16 @@ export const migratePipelineRun = (pipelineRun: PipelineRun): PipelineRun => {
   return newPipelineRun;
 };
 
+export const getPipelineName = (pipeline?: Pipeline, latestRun?: PipelineRun): string => {
+  if (!pipeline && !latestRun) {
+    return null;
+  }
+  return pipeline
+    ? pipeline.metadata.name
+    : latestRun.spec.pipelineRef?.name ??
+        (latestRun.metadata.annotations?.preferredName || latestRun.metadata.name);
+};
```

Comment:
```suggestion
export const getPipelineName = (pipeline?: Pipeline, latestRun?: PipelineRun): string => {
  if (pipeline) {
    return pipeline.metadata.name
  }
  if (pipelineRun) {
    return latestRun.spec.pipelineRef?.name ??
        (latestRun.metadata.annotations?.preferredName || latestRun.metadata.name)
  }
  return null;
};
```
Nit: You could write this in a fashion where it's flatter and easier to compartmentalize the logic.

Not required, just a thought.

Reply:
this does seem to be easier to read and compartmentalize the logic.
updated.